File: assignments/a1/q7/Python/Balanced.py
#Balanced:top#
#Balanced:import#

# <balanced> ::= <para> AT
class Balanced(_Start): #Balanced:class#

    className = "Balanced"
    ruleString = "<balanced> ::= <para> AT"
    para = None

    def __init__(para):
        #Balanced:init#
        self.para = para

#Balanced#

File: assignments/a1/q7/Java/Balanced.java
//Balanced:top//
//Balanced:import//
import java.util.*;

// <balanced> ::= <para> AT
public class Balanced extends _Start /*Balanced:class*/ {

    public static final String $className = "Balanced";
    public static final String $ruleString =
        "<balanced> ::= <para> AT";

    public Para para;

    public Balanced(Para para) {
//Balanced:init//
        this.para = para;
    }

    public static Balanced parse(Scan scn$, Trace trace$) {
        if (trace$ != null)
            trace$ = trace$.nonterm("<balanced>", scn$.lno);
        Para para = Para.parse(scn$, trace$);
        scn$.match(Token.Match.AT, trace$);
        return new Balanced(para);
    }

//Balanced//
}
